lib/core/test/unit/Cardano/Pool/DB/Properties.hs

```diff
@@ -1284,6 +1294,38 @@ prop_MultiPoolCertificateSequence_coverage mpcs = checkCoverage
     certificateSequences = getSinglePoolSequences mpcs
     poolCount = length certificateSequences
 
+
+-- | read . put == pure
+prop_putHeaderListHeader
+    :: DBLayer IO
+    -> [BlockHeader]
+    -> Positive Int
+    -> Property
+prop_putHeaderListHeader DBLayer{..} headers (Positive k) =
+    monadicIO (setup >> prop)
+  where
+    sortedHeaders = sortBy (\a b -> compare (blockHeight a) (blockHeight b))
+        . L.nubBy (\a b -> (blockHeight a) == (blockHeight b)) . nub $ headers
+    expected = reverse . (if k == 0 then id else take k) . reverse $ sortedHeaders
+    setup = run $ atomically cleanDB
+    prop = do
+        void $ run $ atomically $ forM sortedHeaders putHeader
+        lHeaders <- run $ atomically $ listHeaders k
+        monitor $ counterexample $ unlines
+            [ "Read from DB: " <> show lHeaders
+            , "Expected: " <> show expected
+            , "Input: " <> show sortedHeaders
+            , "k: " <> show k]
+        monitor $ classify (null lHeaders) "Empty distributions"
+        assert (lHeaders == expected)
+        assert (strictlyIncreasing . fmap blockHeight $ lHeaders)
+
+    strictlyIncreasing :: Ord a => [a] -> Bool
+    strictlyIncreasing [] = True
+    strictlyIncreasing [_] = True
+    strictlyIncreasing (x:y:xs) = let !b = (x < y) in b && strictlyIncreasing (y:xs)
```

Comment:
I find this less readable.

Reply:
That was indeed just a suggestion.

But I would still appreciate it if you could fix the line length issue. This would work:
```suggestion
    strictlyIncreasing [] = True
    strictlyIncreasing [_] = True
    strictlyIncreasing (x:y:xs) =
        let !b = (x < y) in b && strictlyIncreasing (y:xs)
```

Reply:
Done